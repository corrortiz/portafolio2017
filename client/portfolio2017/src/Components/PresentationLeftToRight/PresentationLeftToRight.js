// @flow
import React, { Component } from 'react';
//HOC global store
import GlobalsConnect from '../../HOC/GlobalsConnect/GlobalsConnect';
//Locale selectors
import { lenguajeSelector } from '../../Store/Actions/globals';
import { WebDev, WebDevText } from '../../Assets/diccionary';
//MUI Controls
import Grid from 'material-ui/Grid';
import { withStyles } from 'material-ui/styles';
//CSS in JS Styles
const styles = theme => ({
  root: {}
});

/**
 * This class represents a common section of two columns
 * one with information and the other with image from left to right
 */
export class PresentationLeftToRight extends Component {
  render() {
    const { lenguaje } = this.props.globals;
    const { classes } = this.props;

    return (
      <Grid
        container
        spacing={8}
        alignItems={'center'}
        justify={'center'}
        alignContent={'center'}
        className={`PresentationLeftToRight box ${classes.root}`}
      >
        {/*Image*/}
        <Grid
          item
          xs={10}
          sm={6}
          md={6}
          lg={6}
          className="PresentationLeftToRight__two"
        >
          <div className="PresentationLeftToRight__two__image AnimationBorder" />
        </Grid>

        {/*Information text*/}
        <Grid item xs={12} lg={6} className="PresentationLeftToRight__one">
          <h1 className="PresentationLeftToRight__one__title">
            {lenguajeSelector(lenguaje, WebDev)}
          </h1>
          <p className="PresentationLeftToRight__one__text">
            {lenguajeSelector(lenguaje, WebDevText)}
          </p>
        </Grid>
      </Grid>
    );
  }
}

PresentationLeftToRight = withStyles(styles)(PresentationLeftToRight);

export default GlobalsConnect(PresentationLeftToRight);
