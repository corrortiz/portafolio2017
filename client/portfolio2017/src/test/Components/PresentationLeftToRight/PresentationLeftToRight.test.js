import React from 'react';
import { shallow } from 'enzyme';
import { PresentationLeftToRight } from '../../../Components/PresentationLeftToRight/PresentationLeftToRight';

describe('PresentationLeftToRight Component', () => {
  it('Should compare a snapshop of the component', () => {
    const wrapper = shallow(<PresentationLeftToRight />);
    expect(wrapper).toMatchSnapshot();
  });
});
