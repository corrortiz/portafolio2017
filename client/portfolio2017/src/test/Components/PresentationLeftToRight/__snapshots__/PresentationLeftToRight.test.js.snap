// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresentationLeftToRight Component Should compare a snapshop of the component 1`] = `
<PresentationLeftToRight
  classes={
    Object {
      "root": "PresentationLeftToRight-root-1",
    }
  }
/>
`;
